src/components/mypage/profile-edit.tsx

```diff
@@ -0,0 +1,108 @@
+'use client';
+import { useState } from 'react';
+import { useForm } from 'react-hook-form';
+import Image from 'next/image';
+
+import CircleEdit from '~/src/assets/icons/circle-edit.svg?url';
+import MyProfileEdit from '~/src/assets/images/mypage-profile-edit.png';
+import Button from '~/src/components/common/button';
+import {
+  Form,
+  FormField,
+  FormItem,
+  FormLabel,
+} from '~/src/components/common/form';
+import Input from '~/src/components/common/input';
+import { DialogFooter } from '~/src/components/common/modal';
+import {
+  Dialog,
+  DialogContent,
+  DialogHeader,
+  DialogTitle,
+  DialogTrigger,
+} from '~/src/components/common/modal';
+import { useEditUser } from '~/src/services/auths/edit-user';
+import { useGetUserInfo } from '~/src/services/auths/get-user';
+import { type UserEditType } from '~/src/services/auths/types';
+
+export default function ProfileEdit() {
+  const { data: user } = useGetUserInfo();
+  const [isModalOpen, setIsModalOpen] = useState(false);
+
+  const form = useForm<UserEditType>({
+    defaultValues: {
+      companyName: user?.companyName,
+      // image: user?.image || '',
+    },
+  });
+
+  const {
+    formState: { isDirty },
+  } = form;
+  const handleCancel = () => {
+    setIsModalOpen(false);
+    form.reset();
+  };
+
+  const mutation = useEditUser();
+
+  const onSubmit = (data: UserEditType) => {
+    mutation.mutate(data);
+    setIsModalOpen(false);
+  };
```

Comment:
```suggestion
   const onSubmit = (data: UserEditType) => {
    mutation.mutate(data, {
      onSuccess: () => handleCancel(),
    });
  };
```

submit 이후에 form에 작성한 내용도 삭제되면 좋을것 같아요. 그리고 비동기 로직 끝난 후에 작동하는게 더 좋을것 같습니다!